contracts/base/AaveV2Wrapper.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../libraries/Errors.sol";
+import "../interfaces/IPoolV2.sol";
+import "../interfaces/IWETHGateway.sol";
+import "../interfaces/LiquidityPoolInterfaces.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+/**
+ * @author CRUIZE.
+ * @title Cruize AaveV2Wrapper.
+ * @notice AaveV2Wrapper contract is an integration with aave(ETHEREUM-MAINNET) 
+ * to borrow a 25% loan against the user's deposited asset.
+ * 1 - User's deposited 90% amount will be deposited into Aave.
+ * 2 - Remaining 10% amount will be transferred cruize wallet
+ * 3 - A 25% loan is borrowed in USDC and transferred to the cruize wallet.
+ */
+contract AaveV2Wrapper is OwnableUpgradeable, ReentrancyGuardUpgradeable {
+    using SafeMath for uint256;
+
+    //----------------------------//
+    //        State Variable      //
+    //----------------------------//
+
+    uint256 public priceFloor;
+    uint256 public constant BASE = 10000;
+
+    address public crContract;
+    address public cruize_wallet;
+    uint256 public borrowRatio = 2500; // 25% of 1000
+    uint256 public constant aaveDefaulLTV = 8250; // 82.5% of 1000
+    uint256 private constant USD_DECIMALS = 8;
+    uint256 private constant ETH_DECIMALS = 18;
+
+    //----------------------------//
+    //        Mappings            //
+    //----------------------------//
+
+    // reserveToken => crToken
+    mapping(address => address) public lpTokens;
+    // reserveToken => oracles
+    mapping(address => address) public oracles;
+
+    //----------------------------//
+    //        Address             //
+    //----------------------------//
+
+    IPoolV2 internal constant POOL =
+        IPoolV2(0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9);
+
+    IWETHGateway internal constant WETHGATEWAY =
+        IWETHGateway(0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04);
+
+    IERC20 internal constant BORROW_ASSET =
+        IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48); // USDC
+
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
```

Comment:
add these in the constants file as well.

Reply:
Sure, that would be fine